test-framework/junit5-component/src/test/java/io/quarkus/test/component/config/ConfigMappingTest.java

```diff
@@ -0,0 +1,56 @@
+package io.quarkus.test.component.config;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.component.QuarkusComponentTest;
+import io.quarkus.test.component.TestConfigProperty;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
+
+@QuarkusComponentTest
+@TestConfigProperty(key = "foo.bar", value = "true")
+public class ConfigMappingTest {
+
+    @Inject
+    Foo foo;
+
+    @TestConfigProperty(key = "foo.oof", value = "boom")
+    @Test
+    public void testMapping() {
+        assertTrue(foo.config.bar());
+        assertEquals("loom", foo.config.baz());
+        assertEquals("boom", foo.config.oof());
+    }
+
+    @TestConfigProperty(key = "foo.oof", value = "boomboom")
+    @Test
+    public void testAnotherMapping() {
+        assertTrue(foo.config.bar());
+        assertEquals("loom", foo.config.baz());
+        assertEquals("boomboom", foo.config.oof());
+    }
+
+    @Singleton
+    public static class Foo {
+
+        @Inject
+        FooConfig config;
+    }
+
+    @ConfigMapping(prefix = "foo")
+    interface FooConfig {
+
+        boolean bar();
+
+        @WithDefault("loom")
+        String baz();
+
+        String oof();
+    }
```

Comment:
I suppose nested config mappings do not need extra care and are handled by Smallrye Config itself? I.e. if you had this:

```
    @ConfigMapping(prefix = "foo")
    interface FooConfig {

        boolean bar();

        @WithDefault("loom")
        String baz();

        String oof();

        Fii fii();

        interface Fii {
            String bar();
        }
    }
```

... then I guess `Fii` would not need to be handled explicitly by your code?

Reply:
That's my understanding too. But I can add a test to make sure it really works ;-).

Reply:
Test added.